frontend/src/component/common/ApiTokenTable/RemoveApiTokenButton/RemoveApiTokenButton.tsx

```diff
@@ -1,72 +1,49 @@
-import {
-    DELETE_API_TOKEN,
-    DELETE_PROJECT_API_TOKEN,
-} from 'component/providers/AccessProvider/permissions';
 import { Delete } from '@mui/icons-material';
 import { styled } from '@mui/material';
-import {
-    IApiToken,
-    useApiTokens,
-} from 'hooks/api/getters/useApiTokens/useApiTokens';
-import AccessContext from 'contexts/AccessContext';
-import { useContext, useState } from 'react';
+import { IApiToken } from 'hooks/api/getters/useApiTokens/useApiTokens';
+import { useState } from 'react';
 import { Dialogue } from 'component/common/Dialogue/Dialogue';
 import useToast from 'hooks/useToast';
-import useApiTokensApi from 'hooks/api/actions/useApiTokensApi/useApiTokensApi';
 import PermissionIconButton from 'component/common/PermissionIconButton/PermissionIconButton';
-import useProjectApiTokensApi from '../../../../hooks/api/actions/useProjectApiTokensApi/useProjectApiTokensApi';
+import { formatUnknownError } from 'utils/formatUnknownError';
 
 const StyledUl = styled('ul')({
     marginBottom: 0,
 });
 
 interface IRemoveApiTokenButtonProps {
     token: IApiToken;
+    permission: string;
+    remove: () => void;
+    track?: () => void;
     project?: string;
 }
 
 export const RemoveApiTokenButton = ({
     token,
+    permission,
+    remove,
+    track,
     project,
 }: IRemoveApiTokenButtonProps) => {
-    const { hasAccess, isAdmin } = useContext(AccessContext);
-    const { deleteToken } = useApiTokensApi();
-    const { deleteToken: deleteProjectToken } = useProjectApiTokensApi();
     const [open, setOpen] = useState(false);
-    const { setToastData } = useToast();
-    const { refetch } = useApiTokens();
+    const { setToastData, setToastApiError } = useToast();
 
-    const permission = Boolean(project)
-        ? DELETE_PROJECT_API_TOKEN
-        : DELETE_API_TOKEN;
+    const onRemove = async () => {
+        try {
+            await remove();
+            setOpen(false);
 
-    const canRemove = () => {
-        if (isAdmin) {
-            return true;
-        }
-        if (token && token.projects && project && permission) {
-            const { projects } = token;
-            for (const tokenProject of projects) {
-                if (!hasAccess(permission, tokenProject)) {
-                    return false;
-                }
+            setToastData({
+                type: 'success',
+                title: 'API token removed',
+            });
+            if (track && typeof track === 'function') {
```

Comment:
does the button need to be coupled to the track fn? what if we passed it as part of the remove that we inject anyway. The smaller the public API the better.